compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1142,9 +1142,9 @@ public void visit(BLangTupleVariable varNode) {
         String name = "$tuple$";
         final BLangSimpleVariable tuple =
                 ASTBuilderUtil.createVariable(varNode.pos, name, symTable.arrayAllType, null,
-                                              new BVarSymbol(0, names.fromString(name), this.env.scope.owner.pkgID,
-                                                             symTable.arrayAllType, this.env.scope.owner, varNode.pos,
-                                                             VIRTUAL));
+                        new BVarSymbol(Flags.DESTRUCTURED, names.fromString(name), this.env.scope.owner.pkgID,
+                                symTable.arrayAllType, this.env.scope.owner, varNode.pos,
+                                VIRTUAL));
```

Comment:
Can we move this varSymbol new instance creation logic to a method and call? Looks like there are 5-6 places the same is happening.

Reply:
Fixed

Reply:
Frontend team has merged a PR which identifies the desugard symbol. So took advantage of it and refactored to use it